src/mgmt/tests/srv_drpc_tests.c

```diff
@@ -1429,6 +1429,81 @@ test_drpc_pool_query_success_rebuild_err(void **state)
 	D_FREE(resp.body.data);
 }
 
+/*
+ * dRPC pool create tests
+ */
+static void
+expect_create_resp_with_error(Drpc__Response *resp, int exp_error)
+{
+	Mgmt__PoolCreateResp	*pc_resp = NULL;
+
+	assert_int_equal(resp->status, DRPC__STATUS__SUCCESS);
+	assert_non_null(resp->body.data);
+
+	pc_resp = mgmt__pool_create_resp__unpack(NULL, resp->body.len,
+						 resp->body.data);
+	assert_non_null(pc_resp);
+	assert_int_equal(pc_resp->status, exp_error);
+
+	mgmt__pool_create_resp__free_unpacked(pc_resp, NULL);
+}
+
+static void
+pack_pool_create_req(Mgmt__PoolCreateReq *req, Drpc__Call *call)
+{
+	size_t	len;
+	uint8_t	*body;
+
+	len = mgmt__pool_create_req__get_packed_size(req);
+	D_ALLOC(body, len);
+	assert_non_null(body);
+
+	mgmt__pool_create_req__pack(req, body);
+
+	call->body.data = body;
+	call->body.len = len;
+}
+
+static void
+test_drpc_pool_create_invalid_acl(void **state)
+{
+	Drpc__Call		call = DRPC__CALL__INIT;
+	Drpc__Response		resp = DRPC__RESPONSE__INIT;
+	Mgmt__PoolCreateReq	pc_req = MGMT__POOL_CREATE_REQ__INIT;
+	size_t			num_acl = 2;
+	size_t			i;
+	char			**bad_acl;
+	const char		*ace = "A::myuser@:rw"; /* to be duplicated */
+
+	pc_req.uuid = TEST_UUID;
+
+	/* Duplicate entries in the ACL should be invalid */
+	D_ALLOC_ARRAY(bad_acl, num_acl);
+	assert_non_null(bad_acl);
+	for (i = 0; i < num_acl; i++) {
+		D_STRNDUP(bad_acl[i], ace, DAOS_ACL_MAX_ACE_STR_LEN);
+		assert_non_null(bad_acl[i]);
+	}
+
+	pc_req.n_acl = num_acl;
+	pc_req.acl = bad_acl;
+
+	pack_pool_create_req(&pc_req, &call);
+
+	ds_mgmt_drpc_pool_create(&call, &resp);
+
+	expect_create_resp_with_error(&resp, -DER_INVAL);
+
+	/* clean up */
+	for (i = 0; i < num_acl; i++) {
+		D_FREE(bad_acl[i]);
+	}
+	D_FREE(bad_acl);
+	D_FREE(call.body.data);
+	D_FREE(resp.body.data);
+
```

Comment:
nit: extra blank line

Reply:
Whoops. I think I will request landing as there is nothing else to fix on this one at the moment. I will come back to fix next time I'm in this area.